erpnext/projects/doctype/project/project.py

```diff
@@ -267,13 +267,129 @@ def update_dependencies_on_duplicated_project(self):
 				task_doc.save()
 
 def get_timeline_data(doctype, name):
-	'''Return timeline for attendance'''
-	return dict(frappe.db.sql('''select unix_timestamp(from_time), count(*)
+	'''returns timeline data based on linked records in dashboard'''
+	from six import iteritems
+	from frappe.utils import get_timestamp
+
+	out = {}
+
+	'''timesheets'''
+	items = dict(frappe.db.sql('''select unix_timestamp(from_time), count(*)
 		from `tabTimesheet Detail` where project=%s
 			and from_time > date_sub(curdate(), interval 1 year)
 			and docstatus < 2
 			group by date(from_time)''', name))
 
+	for date, count in items.iteritems():
+		timestamp = get_timestamp(date)
+		out.update({ timestamp: count })
+
+	'''task'''
+	items = dict(frappe.db.sql('''select creation, count(*)
+		from `tabTask` where project=%s
+			and creation > date_sub(curdate(), interval 1 year)
+			group by creation''', name))
+
+	for date, count in iteritems(items):
+		timestamp = get_timestamp(date)
+		if not timestamp in out:
+			out.update({timestamp: count})
+		else :
+			out.update({timestamp: out[timestamp] + count})
+
+	'''expense claim'''
+	items = dict(frappe.db.sql('''select creation, count(*)
+		from `tabExpense Claim` where project=%s
+			and creation > date_sub(curdate(), interval 1 year)
+			group by creation''', name))
+
+	for date, count in iteritems(items):
+		timestamp = get_timestamp(date)
+		if not timestamp in out:
+			out.update({timestamp: count})
+		else :
+			out.update({timestamp: out[timestamp] + count})
+
+	'''issue'''
+	items = dict(frappe.db.sql('''select creation, count(*)
```

Comment:
also might be a good idea to move this into functions as this query has a standard pattern.